ui/src/datasource/edit/schema/DatasourceSchemaEdit.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <v-form
+    v-model="isValid"
+  >
+    <v-card-actions>
+      <v-select
+        v-model="dataSchema.mode"
+        :items="availableSchemaModes"
+        label="Mode"
+        :rules="[required]"
+      />
+      <v-btn
+        color="primary"
+        class="ma-2"
+        @click="onGenerate"
+      >
+        Generate
+      </v-btn>
+    </v-card-actions>
+    <v-textarea
+      v-model="dataSchema.data"
+      label="Datasource schema suggestion"
+      @keyup="formChanged"
+    />
+  </v-form>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import Component from 'vue-class-component'
+import * as DatasourceREST from './../../datasourceRest'
+
+import { Emit, PropSync } from 'vue-property-decorator'
+import { requiredRule } from '../../../validators'
+import { DataSchema } from '../../datasource'
+
+@Component({ })
+export default class DatasourceSchemaEdit extends Vue {
+  private availableSchemaModes = ['NONE', 'FAST', 'DETAILED']
```

Comment:
Please use this in the model `mode: "NONE" | "FAST" | "DETAILED"' to ensure these are the only viable values by typing